inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -285,3 +295,47 @@ def refextract(obj, eng):
     elif len(text_references) > len(pdf_references):
         obj.log.info('Extracted %d references from text.', len(text_references))
         obj.data['references'] = text_references
+
+
+@with_debug_logging
+def set_document_type_and_refereed(obj, eng):
+    publications = obj.data.get('publication_info')
+    journals = filter(lambda x: 'journal_title' in x, publications)
+
+    modify_doc_type = False
+    modify_ref_f = False
+
+    if not journals:
+        return None
+
+    for index, journal in enumerate(journals):
+        query = RecordMetadata.query.filter(
+            RecordMetadata.json['_collections'].op('?')('Journals')).filter(
+            cast(RecordMetadata.json['short_title'], String) == type_coerce(journal['journal_title'], JSON))
+        result = db.session.execute(query).fetchone()
+
+        if result:
+            if result.records_metadata_json['refereed'] or not result.records_metadata_json['proceedings']:
+                modify_doc_type = True
+
+            if result.records_metadata_json['refereed']:
+                modify_ref_f = True
+
+            if result.records_metadata_json['refereed'] and result.records_metadata_json['proceedings']\
+                    and 'conference paper' not in obj.data.get('document_type'):
+                obj.data['refereed'] = True
+
+            if result.records_metadata_json['refereed'] and not result.records_metadata_json['proceedings']:
+                obj.data['refereed'] = True
+
+    if modify_doc_type:
+        doc_types = obj.data.get('document_type')
+
+        for index_d, type in enumerate(doc_types):
+            if type == 'article':
+                del obj.data['document_type'][index_d]
+                obj.data['document_type'].append('conference paper')
```

Comment:
you can combine these two lines by overwriting instead of deleting and appending: `obj.data[index_d] = 'conference paper'`

Reply:
@michamos i was doing that at first, but then i thought what if there are both 'article' and 'conference paper' in there? I am not sure whether that ever happens tho

Reply:
good point :+1: the first rule of inspire metadata is "anything that can happen will hapen".

Reply:
However, you are not checking if `conference paper` is already present.